docs/markdown/Running-Meson.md

```diff
@@ -4,49 +4,45 @@ short-description: Building a project with Meson
 
 # Running Meson
 
-There are two different ways of invoking Meson. First, you can run it
-directly from the source tree with the command
-`/path/to/source/meson.py`. Meson may also be installed in which case
-the command is simply `meson`. In this manual we only use the latter
-format for simplicity.
+There are two different ways of invoking Meson. First, you can run it directly
+from the source tree with the command `/path/to/source/meson.py`. Meson may
+also be installed in which case the command is simply `meson`. In this manual
+we only use the latter format for simplicity.
 
-Additionally, the invocation can pass options to meson.
-The list of options is documented [here](Builtin-options.md).
+Additionally, the invocation can pass options to meson. The list of options is
+documented [here](Builtin-options.md).
 
-At the time of writing only a command line version of Meson is
-available. This means that Meson must be invoked using the
-terminal. If you wish to use the MSVC compiler, you need to run Meson
-under "Visual Studio command prompt".
+At the time of writing only a command line version of Meson is available. This
+means that Meson must be invoked using the terminal. If you wish to use the
+MSVC compiler, you need to run Meson under "Visual Studio command prompt".
 
-Configuring the source
-==
+## Configuring the build directory
 
-Let us assume that we have a source tree that has a Meson build
-system. This means that at the topmost directory has a file called
-`meson.build`. We run the following commands to get the build started.
+Let us assume that we have a source tree that has a Meson build system. This
+means that at the topmost directory has a file called `meson.build`. We run the
+following commands to get the build started.
 
+```sh
+cd /path/to/source/root
+meson setup ../builddir
+```
```

Comment:
I prefer `builddir` than `../builddir`. Can you eleborate why using the latter?

Reply:
It's to indicate it is an [out of source build](http://voices.canonical.com/jussi.pakkanen/2013/04/16/why-you-should-consider-using-separate-build-directories/). I can do `/tmp/project-x/builddir` if that's any better.

Reply:
I think out-of-source build means that the generated object files is laying in directories other than where the source file is. So using `builddir` is enough.

Reply:
Oh OK, updated now.

For me it's a distraction having build directories appear as untracked files in git, but hey ho, each to their own. It's a trivial change. Done now.

Reply:
OK, clearly there is a fundamentally different way of working here.

The guide goes on to give an example of building with clang, instead of the default system compiler, using a `buildclang` directory. The point there is that creating an alternate build is trivial. Adding that to `.gitignore` adds an extra step and becomes awkward when the temporary build directory changes name. That's why I don't tend to add to .gitignore now. `git add -u` works fine so long as new project files aren't created.

The idea behind ../ was for anyone following the basic tutorial to ask themselves 'where is my build'. Then they discover it is just outside of their source directory. That should be an "aha" moment. They then understand that the build directory can be any where they want, not just inside their source tree. Maybe for their own personal projects having `builddir` inside the source tree is fine. It's one directory. When they start trying alternate builds, with different compilers, build options, cross-compiling, packaging, etc. then it might be better to use an external build area. That is then easily deleted when work is complete. That isn't discussed on that page.

Although there are only three characters, `../`, involved in this discussion there was thought put in to how the audience for the document would react to the idea.